doc/modules/tree.rst

```diff
@@ -429,99 +433,109 @@ Mathematical formulation
 
-Recurse for subsets :math:`Q_{left}(\theta^*)` and
-:math:`Q_{right}(\theta^*)` until the maximum allowable depth is reached,
+Recurse for subsets :math:`Q_m^{left}(\theta^*)` and
+:math:`Q_m^{right}(\theta^*)` until the maximum allowable depth is reached,
 :math:`N_m < \min_{samples}` or :math:`N_m = 1`.
 
 Classification criteria
 -----------------------
 
 If a target is a classification outcome taking on values 0,1,...,K-1,
-for node :math:`m`, representing a region :math:`R_m` with :math:`N_m`
-observations, let
+for node :math:`m`, let
 
 .. math::
 
-    p_{mk} = 1/ N_m \sum_{x_i \in R_m} I(y_i = k)
+    p_{mk} = 1/ N_m \sum_{y \in Q_m} I(y = k)
 
-be the proportion of class k observations in node :math:`m`
+be the proportion of class k observations in node :math:`m`. If :math:`m` is a
+terminal node, `predict_proba` for this region is set to :math:`p_{mk}`.
+Common measures of impurity are the following.
 
-Common measures of impurity are Gini
+Gini:
 
 .. math::
 
-    H(X_m) = \sum_k p_{mk} (1 - p_{mk})
+    H(Q_m) = \sum_k p_{mk} (1 - p_{mk})
 
-Entropy
+Entropy:
 
 .. math::
 
-    H(X_m) = - \sum_k p_{mk} \log(p_{mk})
+    H(Q_m) = - \sum_k p_{mk} \log(p_{mk})
 
-and Misclassification
+Misclassification:
 
 .. math::
 
-    H(X_m) = 1 - \max(p_{mk})
-
-where :math:`X_m` is the training data in node :math:`m`
+    H(Q_m) = 1 - \max(p_{mk})
 
 Regression criteria
 -------------------
 
-If the target is a continuous value, then for node :math:`m`,
-representing a region :math:`R_m` with :math:`N_m` observations, common
-criteria to minimise as for determining locations for future
-splits are Mean Squared Error, which minimizes the L2 error
-using mean values at terminal nodes, and Mean Absolute Error, which
-minimizes the L1 error using median values at terminal nodes.
+If the target is a continuous value, then for node :math:`m`, common
+criteria to minimize as for determining locations for future splits are Mean
+Squared Error (MSE or L2 error), Poisson deviance as well as Mean Absolute
+Error (MAE or L1 error). MSE and Poisson deviance both set the predicted value
+of terminal nodes to the learned mean value :math:`\bar{y}_m` of the node
+whereas the MAE sets the predicted value of terminal nodes to the median
+:math:`median(y)_m`.
 
 Mean Squared Error:
 
 .. math::
 
-    \bar{y}_m = \frac{1}{N_m} \sum_{i \in N_m} y_i
+    \bar{y}_m = \frac{1}{N_m} \sum_{y \in Q_m} y
 
-    H(X_m) = \frac{1}{N_m} \sum_{i \in N_m} (y_i - \bar{y}_m)^2
+    H(Q_m) = \frac{1}{N_m} \sum_{y \in Q_m} (y - \bar{y}_m)^2
 
-Mean Absolute Error:
+Half Poisson deviance:
 
 .. math::
 
-    median(y)_m = \underset{i \in N_m}{\mathrm{median}}(y_i)
+    H(Q_m) = \frac{1}{N_m} \sum_{y \in Q_m} (y \log\frac{y}{\bar{y}_m}
+    - y + \bar{y}_m)
+
+Setting `criterion="poisson"` might be a good choice if your target is a count
+or a frequency (count per some unit). In any case, :math:`y >= 0` is a
+necessary condition to use this criterion.
```

Comment:
I think we should document that it may be much slower than MSE.
Maybe we can make the same comment about MAE (it's much slower than MSE)